plugins/aws/sts_provisioner.go

```diff
@@ -254,7 +256,7 @@ type assumeRoleProvider struct {
 }
 
 func (p assumeRoleProvider) Retrieve(ctx context.Context) (aws.Credentials, error) {
-	credentials, err := p.AssumeRoleProvider.Retrieve(ctx)
+	credentials, err := ExecuteSilently(p.AssumeRoleProvider.Retrieve)(ctx)
```

Comment:
Let's add the execute silently in the provision method instead here: `tempCredentials, err := tempCredentialsProvider.Retrieve(ctx)` and remove it from assume role and session providers levels. This is because of an edge case mentioned in the other comment's updated section